File: frontend/apps/android/app/src/main/java/app/pantopus/android/ui/screens/mailbox/disambiguate/components/MatchBadge.kt
@file:Suppress("MagicNumber", "PackageNaming")

package app.pantopus.android.ui.screens.mailbox.disambiguate.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import app.pantopus.android.ui.screens.mailbox.disambiguate.MailMatchTier
import app.pantopus.android.ui.theme.PantopusColors
import app.pantopus.android.ui.theme.Radii
import app.pantopus.android.ui.theme.Spacing

/**
 * A13.15 Disambiguate — tiny uppercase badge tagging a candidate's
 * OCR-vs-record match strength: Strong match (success), Partial (amber),
 * Weak (neutral). Mirrors the iOS `MatchBadge`.
 */
@Composable
fun MatchBadge(
    tier: MailMatchTier,
    percent: Int,
    modifier: Modifier = Modifier,
) {
    val palette = matchPalette(tier)
    Box(
        modifier =
            modifier
                .clip(RoundedCornerShape(Radii.xs))
                .background(palette.background)
                .border(1.dp, palette.border, RoundedCornerShape(Radii.xs))
                .padding(horizontal = Spacing.s2, vertical = 2.dp)
                .semantics { contentDescription = "${tier.word}, $percent percent match" },
    ) {
        Text(
            text = "${tier.word} · $percent%".uppercase(),
            style =
                TextStyle(
                    fontSize = 9.5.sp,
                    fontWeight = FontWeight.Bold,
                    letterSpacing = 0.1.sp,
                ),
            color = palette.foreground,
        )
    }
}

private data class MatchPalette(val background: Color, val foreground: Color, val border: Color)

private fun matchPalette(tier: MailMatchTier): MatchPalette =
    when (tier) {
        MailMatchTier.Strong ->
            MatchPalette(PantopusColors.successBg, PantopusColors.success, PantopusColors.successLight)
        MailMatchTier.Partial ->
            MatchPalette(PantopusColors.warmAmberBg, PantopusColors.warmAmber, PantopusColors.warningLight)
        MailMatchTier.Weak ->
            MatchPalette(PantopusColors.appSurfaceSunken, PantopusColors.appTextSecondary, PantopusColors.appBorder)
    }

File: frontend/apps/ios/Pantopus/Features/Mailbox/Disambiguate/Components/MatchBadge.swift
//
//  MatchBadge.swift
//  Pantopus
//
//  A13.15 Disambiguate — tiny uppercase badge tagging a candidate's
//  OCR-vs-record match strength: Strong match (success), Partial (amber),
//  Weak (neutral). Mirrors the Android `MatchBadge`.
//

import SwiftUI

/// Match-strength badge rendered beside a candidate's name.
@MainActor
struct MatchBadge: View {
    let tier: MailMatchTier
    let percent: Int

    var body: some View {
        Text("\(tier.word) · \(percent)%")
            .font(.system(size: 9.5, weight: .bold))
            .tracking(0.1)
            .textCase(.uppercase)
            .foregroundStyle(palette.foreground)
            .padding(.horizontal, Spacing.s2)
            .padding(.vertical, 2)
            .background(palette.background)
            .clipShape(RoundedRectangle(cornerRadius: Radii.xs, style: .continuous))
            .overlay(
                RoundedRectangle(cornerRadius: Radii.xs, style: .continuous)
                    .stroke(palette.border, lineWidth: 1)
            )
            .accessibilityLabel("\(tier.word), \(percent) percent match")
    }

    private struct Palette {
        let background: Color
        let foreground: Color
        let border: Color
    }

    private var palette: Palette {
        switch tier {
        case .strong:
            Palette(
                background: Theme.Color.successBg,
                foreground: Theme.Color.success,
                border: Theme.Color.successLight
            )
        case .partial:
            Palette(
                background: Theme.Color.warmAmberBg,
                foreground: Theme.Color.warmAmber,
                border: Theme.Color.warningLight
            )
        case .weak:
            Palette(
                background: Theme.Color.appSurfaceSunken,
                foreground: Theme.Color.appTextSecondary,
                border: Theme.Color.appBorder
            )
        }
    }
}

#Preview("Match badges") {
    VStack(alignment: .leading, spacing: Spacing.s3) {
        MatchBadge(tier: .strong, percent: 97)
        MatchBadge(tier: .partial, percent: 41)
        MatchBadge(tier: .weak, percent: 22)
    }
    .padding(Spacing.s4)
    .background(Theme.Color.appBg)
}
